shared/src/main/scala/fr/hmil/roshttp/SimpleHttpResponse.scala

```diff
@@ -0,0 +1,59 @@
+package fr.hmil.roshttp
+
+import java.nio.ByteBuffer
+import java.util.concurrent.TimeUnit
+
+import monifu.reactive.Observable
+
+import monifu.concurrent.Implicits.globalScheduler
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * An HTTP response obtained via an [[HttpRequest]]
+ */
+class SimpleHttpResponse(
+    val statusCode: Int,
+    val body: String,
+    val headers: HeaderMap[String])
+  extends HttpResponse
+
+object SimpleHttpResponse extends HttpResponseFactory[SimpleHttpResponse] {
+  override def apply(
+      statusCode: Int,
+      bodyStream: Observable[ByteBuffer],
+      headers: HeaderMap[String])
+      (implicit ec: ExecutionContext, config: HttpConfig): Future[SimpleHttpResponse] = {
+
+    val charset = HttpUtils.charsetFromContentType(headers.getOrElse("content-type", null))
+
+    bodyStream
+      .buffer(FiniteDuration(config.bodyCollectTimeout, TimeUnit.SECONDS))
+      .map({ seq =>
+        // Allocate maximum expected body length
+        val buffer = ByteBuffer.allocate(seq.length * config.streamChunkSize)
+        val totalBytes = seq.foldLeft(0)({(count, chunk) =>
+          buffer.put(chunk)
+          count + chunk.limit
+        })
+        buffer.limit(totalBytes)
+        getStringFromBuffer(buffer, charset)
+      })
+      // TODO: We might not want to depend on monifu's Scheduler
+      .asFuture
```

Comment:
The thing is that both the batch and streaming APIs are backed by the same stream-based backend call (which is what this piece of code is about). This avoids code duplication, reduces the risks of bugs and does probably have little effect on performance for basic usages. Therefore, a Scheduler is required even for the batch API.